sklearn/ensemble/gradient_boosting.py

```diff
@@ -1846,3 +1853,467 @@ def apply(self, X):
         leaves = super(GradientBoostingRegressor, self).apply(X)
         leaves = leaves.reshape(X.shape[0], self.estimators_.shape[0])
         return leaves
+
+
+class GradientBoostingClassifierCV(BaseEstimator):
+    """ Gradient Boosting Classification with Cross Validation
+
+    This class keeps adding estimators to the underlying GB estimator till
+    adding a new estimators does not affect the score by much. The goal of this
+    class is to determine the best choice of `n_estimators` for each set of
+    parameters. In most cases, this will be much faster than using
```

Comment:
When the `n_estimators` provided by the user is smaller than the `n_estimators` required till convergence.